app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js

```diff
@@ -6,6 +6,8 @@ import {StageConstants} from 'core/pipeline/config/stages/stageConstants';
 
 const angular = require('angular');
 
+const excludedCloudProviders = [{ cloudProvider: 'kubernetes', providerVersion: 'v2' }];
```

Comment:
I know you just refactored this bit, but why is there anything referencing kubernetes in the core module?

Reply:
Ah, tbh I spent a bunch of time trying to figure out how to exclude a provider version from a stage type but didn't actually give any consideration to the fact I was polluting the core module with provider-specific code.  I'll go back and see if there's a different way to approach this to keep the kubernetes exclusion limited to the k8s module.

Reply:
I've refactored so that providers now specify stages to exclude in their own module rather than stages excluding providers in the core module.